src/Sulu/Bundle/CategoryBundle/Category/CategoryManager.php

```diff
@@ -235,246 +236,177 @@ public function getFieldDescriptors($locale)
     }
 
     /**
-     * Returns categories with a given parent and/or a given depth-level
-     * if no arguments passed returns all categories.
-     *
-     * @param int         $parent    the id of the parent to filter for
-     * @param int         $depth     the depth-level to filter for
-     * @param string|null $sortBy    column name to sort the categories by
-     * @param string|null $sortOrder sort order
-     *
-     * @return CategoryEntity[]
+     * {@inheritdoc}
      */
-    public function find($parent = null, $depth = null, $sortBy = null, $sortOrder = null)
+    public function findById($id, $locale)
     {
-        return $this->categoryRepository->findCategories($parent, $depth, $sortBy, $sortOrder);
-    }
+        if (!$entity = $this->categoryRepository->findCategoryById($id)) {
+            throw new CategoryNotFoundException();
+        }
 
-    /**
-     * Returns the children for a given category.
-     *
-     * @param int         $key       the key of the category to search the children for
-     * @param string|null $sortBy    column name to sort by
-     * @param string|null $sortOrder sort order
-     *
-     * @return CategoryEntity[]
-     */
-    public function findChildren($key, $sortBy = null, $sortOrder = null)
-    {
-        return $this->categoryRepository->findChildren($key, $sortBy, $sortOrder);
+        return $this->getApiObject($entity, $locale);
     }
 
     /**
-     * Returns a category with a given id.
-     *
-     * @param int $id the id of the category
-     *
-     * @return CategoryEntity
+     * {@inheritdoc}
      */
-    public function findById($id)
+    public function findByKey($key, $locale)
     {
-        return $this->categoryRepository->findCategoryById($id);
+        if (!$entity = $this->categoryRepository->findCategoryByKey($key)) {
+            throw new CategoryNotFoundException();
+        }
+
+        return $this->getApiObject($entity, $locale);
     }
 
     /**
-     * Returns a category with a given key.
-     *
-     * @param string $key the key of the category
-     *
-     * @return CategoryEntity
+     * {@inheritdoc}
      */
-    public function findByKey($key)
+    public function findByIds(array $ids, $locale)
     {
-        return $this->categoryRepository->findCategoryByKey($key);
+        $entities = $this->categoryRepository->findCategoryByIds($ids);
+
+        return $this->getApiObjects($entities, $locale);
     }
 
     /**
-     * Returns the categories with the given ids.
-     *
-     * @param array $ids
-     *
-     * @return CategoryEntity[]
+     * {@inheritdoc}
      */
-    public function findByIds(array $ids)
+    public function find($locale, $parentId = null)
     {
-        return $this->categoryRepository->findCategoryByIds($ids);
+        if ($parentId && !$this->categoryRepository->findCategoryById($parentId)) {
```

Comment:
Maybe we can rewrite the entire query to get this working? I'll have quick look.